include/sys/rwlock.h

```diff
@@ -134,26 +134,40 @@ RW_LOCK_HELD(krwlock_t *rwp)
 	return spl_rwsem_is_locked(SEM(rwp));
 }
 
+#ifdef DEBUG_LOCK_TRACKING
+#define	RWL1(x) #x
+#define	RWL2(x) ML1(x)
+#define	RWLOCK_LOC __FILE__ ":" RWL2(__LINE__)
+
+void spl_rwlock_track_init(krwlock_t *, char *);
+void spl_rwlock_track_destroy(krwlock_t *, char *);
+#else
+#define	spl_rwlock_track_init(a,b)	do {} while (0)
+#define	spl_rwlock_track_destroy(a,b)	do {} while (0)
+#endif
+
 /*
  * The following functions must be a #define and not static inline.
  * This ensures that the native linux semaphore functions (down/up)
  * will be correctly located in the users code which is important
  * for the built in kernel lock analysis tools
  */
 #define rw_init(rwp, name, type, arg)					\
-({									\
```

Comment:
Please, don't do this. You either `({ })` or `do { } while (0)`. There are old code that only use `{ }` but you shouldn't add any more, and definitely not reverting it.